embedder/src/drm_backend.rs

```diff
@@ -917,3 +928,39 @@ pub type GbmDrmCompositor = DrmCompositor<
     Option<OutputPresentationFeedback>,
     DrmDeviceFd,
 >;
+
+fn get_mode_id_for_monitor_from_file(output_name: &str) -> Option<usize> {
+    let path = std::env::var("XDG_CONFIG_HOME")
+        .unwrap_or_else(|_| std::env::var("HOME").unwrap() + "/.config")
+        + "/veshell/persistence/Monitor/"
+        + output_name
+        + ".json";
+
+    info!("path: {}", path);
+
+    let file = match std::fs::File::open(path) {
+        Ok(file) => file,
+        Err(err) => {
+            error!("Failed to open file: {}", err);
+            return None;
+        }
+    };
```

Comment:
You can also use let-else patterns FYI, but a match works too.
https://doc.rust-lang.org/beta/rust-by-example/flow_control/let_else.html